File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/FactoryMethod/Factory/HtmlDialog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CreationalDesignPattern.Refactoring.FactoryMethod
{
    public class HtmlDialog : Dialog
    {
        public override IButton CreateButton()
        {
            return new HtmlButton();
        }
    }
}

File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/factorymethod/factory/HtmlDialog.java
package com.ute.refactoring.factorymethod.factory;

import com.ute.refactoring.factorymethod.buttons.Button;
import com.ute.refactoring.factorymethod.buttons.HtmlButton;

public class HtmlDialog extends Dialog{
    @Override
    public Button createButton() {
        return new HtmlButton();
    }
}
